react-app/src/app/screens/Game/index.js

```diff
@@ -1,22 +1,62 @@
 import React, { Component } from 'react';
 
-import Board from './components/Board';
-import style from './styles.scss';
+import { calculateWinner } from './utils';
+import Game from './layout.js';
+
+class GameContainer extends Component {
+  state = {
+    history: [
+      {
+        squares: Array(9).fill(null)
+      }
+    ],
+    stepNumber: 0,
+    xIsNext: true
+  };
+
+  handleClick = squarePosition => {
+    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[history.length - 1];
+    const squares = [...current.squares];
+    const hasWinner = calculateWinner(squares);
+    const squareIsFilled = squares[squarePosition];
+    if (hasWinner || squareIsFilled) return;
+    squares[squarePosition] = this.state.xIsNext ? 'X' : 'O';
+    this.setState({
+      history: history.concat([
+        {
+          squares
+        }
+      ]),
+      xIsNext: !this.state.xIsNext,
+      stepNumber: history.length
+    });
+  };
+
+  jumpTo = step => {
+    this.setState({
+      stepNumber: step,
+      xIsNext: step % 2 === 0
+    });
+  };
 
-class Game extends Component {
   render() {
+    const history = this.state.history;
```

Comment:
You can define a function to get all this const. `const { current, winner, history } = this.getConst();`